src/Swashbuckle.AspNetCore.SwaggerGen/SchemaGenerator/SchemaGenerator.cs

```diff
@@ -321,7 +321,12 @@ private OpenApiSchema CreateArraySchema(DataContract dataContract, SchemaReposit
             {
                 Type = "array",
                 Items = GenerateSchema(dataContract.ArrayItemType, schemaRepository),
-                UniqueItems = hasUniqueItems ? (bool?)true : null
+                UniqueItems = hasUniqueItems ? (bool?)true : null,
+                Xml = new OpenApiXml()
```

Comment:
Having familiarised myself with the original issue, I think while these might be reasonable defaults to assume, they might not match how the response is actually returned to the user.

I think we're missing a way for the user to customise these or we should infer them from attributes on the type.

It's been a while since I did some XML serialization, but there should be some attributes that can be used to influence how objects are serialised (say to use camelCase not PascalCase) and we should respect those if the user's model uses them.

Reply:
Thanks for the feedback! I'm going to investigate it  🤓.